messagingpush/src/main/java/io/customer/messagingpush/processor/PushMessageProcessorImpl.kt

```diff
@@ -85,4 +94,92 @@ internal class PushMessageProcessorImpl(
             )
         }
     }
+
+    override fun processNotificationClick(activityContext: Context, intent: Intent) {
+        kotlin.runCatching {
+            val payload: CustomerIOParsedPushPayload? =
+                intent.extras?.parcelable(NotificationClickReceiverActivity.NOTIFICATION_PAYLOAD_EXTRA)
+            if (payload == null) {
+                logger.error("Payload is null, cannot handle notification intent")
+            } else {
+                processNotificationIntent(activityContext, payload)
+            }
+        }.onFailure { ex ->
+            logger.error("Failed to process notification intent: ${ex.message}")
+        }
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun processNotificationIntent(activityContext: Context, payload: CustomerIOParsedPushPayload) {
+        trackMetrics(payload)
+        handleDeepLink(activityContext, payload)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun trackMetrics(payload: CustomerIOParsedPushPayload) {
+        if (moduleConfig.autoTrackPushEvents) {
```

Comment:
Is there a chance that `trackMetrics()` would be called manually? If so, the `if (moduleConfig.autoTrackPushEvents)` should be moved into maybe `processNotificationIntent` function.

Reply:
I don't want to pass this responsibility to `processNotificationIntent` actually, because I think `trackMetrics` should be responsible for checking this.

However, I rechecked and realized that latest versions of tests don't require these methods to be exposed, so I have updated the visibility. Also, looking at the code again, I feel like the method names should have been more descriptive after they were moved to `PushMessageProcessorImpl`, I have pushed changes after renaming them so they make more sense.